webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js

```diff
@@ -65,6 +65,42 @@ export class SakaiGrader extends graderRenderingMixin(gradableDataMixin(SakaiEle
     if (typeof MathJax !== "undefined") {
       MathJax.Hub.Queue([ "Typeset", MathJax.Hub ]);
     }
+
+    window.addEventListener(
+      "message",
+      e => {
+        // Will this ever actually be a string?
+        const message = (typeof e.data === "string") ? JSON.parse(e.data) : e.data;
+        if ( message.subject !== "lti.gradeChangeNotify" ) return;
```

Comment:
```suggestion
        if ( (typeof message === "string" && message !== GRADE_CHANGE_NOTIFY) || message.subject !==  GRADE_CHANGE_NOTIFY) return;
```

Reply:
You'll need to add GRADE_CHANGE_NOTIFY = "lti.gradeChangeNotify"; to sakai-grader-constants.js and import it in SakaiGrader.js

Reply:
Thanks - the "will this ever be a string?" is yes, maybe - tools may send JSON or may send a string.  It can change from tool to tool so this is a compatibility fallback.